flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -739,40 +749,59 @@ public void collect(StreamRecord<T> record) {
 
 	static class BroadcastingOutputCollector<T> implements WatermarkGaugeExposingOutput<StreamRecord<T>> {
```

Comment:
Since you are modifying and increasing those classes in size, I would suggest to move them out of the `OperatoChain` (`OperatorChain` is already a bit too big). 

I have already implemented just that as a part of another independent effort. Could you cherry pick this commit of mine:
https://github.com/pnowojski/flink/commit/368de7dd0fb041e1231220d91434cc3a7a07f147

and base your change on it?

Reply:
Okay, I will rebase on it.